MilitaryManager/client-app/src/app/shared/services/api/unit.service.ts

```diff
@@ -4,12 +4,23 @@ import { HttpService } from "../core/http.service";
 import { BaseService } from "../core/base.service";
 import { ClientConfigurationService } from "../core/client-configuration.service";
 import { ServiceType } from "../core/serviceType";
+import {Observable } from "rxjs";
 
-@Injectable()
+@Injectable({
+  providedIn: 'root',
+})
 export class UnitsService extends BaseService<any> {
+  
+  http: HttpService;
   constructor(
     httpService: HttpService,
     configService: ClientConfigurationService) {
     super(httpService, 'weatherForecast', configService, UnitModel, ServiceType.units);
+    this.http = httpService;
   }
+
+  getAll(): Observable<UnitModel[]> {
```

Comment:
No need to create `getAll()` method, it is already implemented in BaseService. As a result not need for 14th and 19th lines